judge/judgedaemon.main.php

```diff
@@ -838,14 +842,16 @@ function fetch_executable_internal(
                     $run_config['hash']
                 );
                 if (isset($error)) {
-                    // FIXME
-                    continue;
+                    $scriptId = $judgeTask['run_script_id'];
+                    disable('judgehost', 'hostname', $myhost, "Retrieving/storing debug script '$scriptId' failed.");
                 }
 
                 $debug_cmd = implode(' ', array_map('dj_escapeshellarg',
                     [$runpath, $workdir, $tmpfile]));
                 system($debug_cmd, $retval);
-                // FIXME: check retval
+                if ($retval !== 0) {
+                    disable('judgehost', 'hostname', $myhost, "Running '$runpath' failed.");
```

Comment:
I am wondering (less strong opinion than above) what the right behavior is here: if executing a debug task fails, it is either a problem with the script or something has been cleaned up. I think I would disable the script for now.